sklearn/utils/estimator_checks.py

```diff
@@ -1190,6 +1189,32 @@ def check_estimators_pickle(name, estimator_orig):
         unpickled_result = getattr(unpickled_estimator, method)(X)
         assert_allclose_dense_sparse(result[method], unpickled_result)
 
+    # check if input with nans pickles properly
+    if name in ALLOW_NAN:
+        # set random 10 elements to be np.nan
+        rng = np.random.RandomState(42)
+        mask = rng.choice(X.size, 10, replace=False)
+        X.reshape(-1)[mask] = np.nan
+        estimator.fit(X, y)
+
+        nan_result = dict()
+        for method in check_methods:
+            if hasattr(estimator, method):
+                nan_result[method] = getattr(estimator, method)(X)
+
+        # pickle and unpickle!
+        pickled_estimator = pickle.dumps(estimator)
+        unpickled_estimator = pickle.loads(pickled_estimator)
+
+        for method in nan_result:
+            unpickled_nan_result = getattr(unpickled_estimator, method)(X)
+            if name == 'ChainedImputer':
+                tol = {'rtol': 1e-5, 'atol': 0.1}
```

Comment:
I'm surprised this is necessary. Don't we get fairly exact equality after pickling, if `random_state` is fixed? Ahh... I see what it is. Awkward:

If we do the pickling before we run each of the methods to get the benchmark result, then we'll be using the same `random_state_`. If we, as the current code does, get the benchmark before the pickling, then the `random_state_` will be affected by the methods.

Reply:
that makes sense. I think we need to run the methods to get the benchmark result first, as in the case of the imputers this fits them. (i.e.  `ChainedImputer.fit()` just calls to `ChainedImputer.fit_transform()`). So I think this extra tolerance is needed